packages/cojson/src/coValues/coPlainText.ts

```diff
@@ -0,0 +1,136 @@
+import { CoValueCore } from "../coValueCore.js";
+import { JsonObject } from "../jsonValue.js";
+import {
+  DeletionOpPayload,
+  InsertionOpPayload,
+  OpID,
+  RawCoList,
+} from "./coList.js";
+
+export type StringifiedOpID = string & { __stringifiedOpID: true };
+
+export function stringifyOpID(opID: OpID): StringifiedOpID {
+  return `${opID.sessionID}:${opID.txIndex}:${opID.changeIdx}` as StringifiedOpID;
+}
+
+type PlaintextIdxMapping = {
+  opIDbeforeIdx: OpID[];
+  opIDafterIdx: OpID[];
+  idxAfterOpID: { [opID: StringifiedOpID]: number };
+  idxBeforeOpID: { [opID: StringifiedOpID]: number };
+};
+
+const segmenter = new Intl.Segmenter("en", { granularity: "grapheme" });
+
+export class RawCoPlainText<
+  Meta extends JsonObject | null = JsonObject | null,
+> extends RawCoList<string, Meta> {
```

Comment:
IMHO we shouldn't rely on inheritance here.

`RawCoList` is already quite complex, extending it makes `RawCoPlainText` practically unmaintainable and harder to use.

What if we move the shared bits into a separate class and reuse it across the two primitives?